src/options/pages/settings/FactoryResetSettings.tsx

```diff
@@ -15,52 +15,116 @@
  * along with this program.  If not, see <http://www.gnu.org/licenses/>.
  */
 
-import { Button, Card } from "react-bootstrap";
+import { Card } from "react-bootstrap";
 import React from "react";
-import { connect } from "react-redux";
+import { useDispatch, useSelector } from "react-redux";
 import notify from "@/utils/notify";
 import extensionsSlice from "@/store/extensionsSlice";
 import servicesSlice from "@/store/servicesSlice";
 import { clearPackages } from "@/baseRegistry";
+import { recipesSlice } from "@/recipes/recipesSlice";
+import { authSlice } from "@/auth/authSlice";
+import { clearLogs, reactivateEveryTab } from "@/background/messenger/api";
+import blueprintsSlice from "@/options/pages/blueprints/blueprintsSlice";
+import settingsSlice from "@/store/settingsSlice";
+import workshopSlice from "@/store/workshopSlice";
+import { sessionChangesActions } from "@/store/sessionChanges/sessionChangesSlice";
+import AsyncButton from "@/components/AsyncButton";
+import { reportEvent } from "@/telemetry/events";
+import { useModals } from "@/components/ConfirmationModal";
+import { type Permissions } from "webextension-polyfill";
+import { selectAdditionalPermissionsSync } from "webext-additional-permissions";
+import { selectSessionId } from "@/pageEditor/slices/sessionSelectors";
+import { persistor } from "@/store/optionsStore";
 
 const { resetOptions } = extensionsSlice.actions;
 const { resetServices } = servicesSlice.actions;
+const { resetRecipes } = recipesSlice.actions;
+const { resetAuth } = authSlice.actions;
+const { resetScreen: resetBlueprintsScreen } = blueprintsSlice.actions;
+const { resetSettings } = settingsSlice.actions;
+const { resetWorkshop } = workshopSlice.actions;
 
-const FactoryResetSettings: React.FunctionComponent<{
-  resetOptions: () => void;
-}> = ({ resetOptions }) => (
-  <Card border="danger">
-    <Card.Header className="danger">Factory Reset</Card.Header>
-    <Card.Body className="text-danger">
-      <p className="card-text">
-        Click here to reset your local PixieBrix data.{" "}
-        <b>This will delete any bricks you&apos;ve installed.</b>
-      </p>
-      <Button
-        variant="danger"
-        onClick={async () => {
-          try {
-            resetOptions();
-            await browser.contextMenus.removeAll();
-            await clearPackages();
-            notify.success("Reset all options and service configurations");
-          } catch (error) {
-            notify.error({
-              message: "Error resetting options and service configurations",
-              error,
+async function revokeAllAdditionalPermissions() {
+  const permissions: Permissions.AnyPermissions =
+    await browser.permissions.getAll();
+  const additional = selectAdditionalPermissionsSync(permissions);
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any -- typings appear to be incorrect
+  await browser.permissions.remove(additional as any);
```

Comment:
You could cast it to the unwrapped type here instead of using `any`, and then you don't have to suppress lint either (or cast in the previous line where the variable is created):

```suggestion
  await browser.permissions.remove(additional as Permissions.Permissions);
```